File: w2/home/lcm/lcm.c
#include <stdio.h>

int gcd(int a, int b) {
    if ( b == 0 ) {
        return a;
    }
    return gcd(b, a%b);
}

int main() {
    FILE *in = fopen("task.in", "r");
    FILE *out = fopen("task.out", "w");
    int a, b, lcm;
    
    fscanf(in, "%d %d", &a, &b);
    
    lcm = a / gcd(a, b) * b ;
    fprintf(out, "%d\n", lcm);
    fclose(in);
    fclose(out);
    
    return 0;
}

File: w4/php_dz/lcm/lcm.php
<?php
    function gcd($a, $b) {
        if ( $b == 0 ) {
            return $a;
        }
        return gcd($b, $a % $b);
    }

    $in = fopen("task.in", "r");
    $out = fopen("task.out", "w");

    // $a = trim(fread($in, 1));
    // $b = fread($in, 2);
    fgets($in, "%d %d", &$a, &$b);

    var_dump($b);

    $lcm = $a / gcd($a, $b) * $b;
    fwrite($out, $lcm . PHP_EOL);

    fclose($in);
    fclose($out);
?>